### PR TITLE
Canal: Clear table cache  when the table structure changed

### PR DESCRIPTION
**Pull request by iyidan**
module Canal:
When received a replication.QueryEvent, check if it is ALTER TABLE Query. 
If true, clear the table cache

### CODE REVIEW

File: canal/sync.go [hunk 1]
```diff
@@ -5,12 +5,18 @@ import (
...
+	"regexp"
+
 	"github.com/juju/errors"
 	"github.com/ngaut/log"
 	"github.com/siddontang/go-mysql/mysql"
 	"github.com/siddontang/go-mysql/replication"
 )
 
+var (
+	expAlterTable = regexp.MustCompile("^ALTER TABLE\\s.*?`([^`\\.]+?)`\\s.*")
```
Comment (siddontang) [inline canal/sync.go]: Hi @iyidan 

Could you add some test for this, like 
```
alter table a.b
alter table `a.b`
alter table b
alter table `b`
```

we can all get table `b` here.

**Comment by siddontang** [comment]
CI failed @iyidan 

Btw, I see this PR contains more than one features, could you split this PR into small ones? So we can review it one by one easily.

File: canal/sync.go (hunk 1 above)
Comment (iyidan) [inline canal/sync.go]: Thanks. Would add it latter.

**Comment by siddontang** [comment]
@iyidan 

Why close it?

**Comment by iyidan** [comment]
@siddontang 
Because the master branch is custom  for my own projects，so i create a new branch to pull request

Please see this new pull request  #113

**Comment by siddontang** [comment]
Thanks @iyidan 
Can we send me the following PR for `SetStartPosition` now ?

**Pull request closed without merging**